elastic/service.go

```diff
@@ -804,7 +804,7 @@ func (s *service) projectSlugToIndexPatterns(projectSlug string, dataSourceTypes
 	patternRoot = "sds-" + strings.Replace(patternRoot, "/", "-", -1) + "-"
 	for _, dataSourceType := range dataSourceTypes {
 		dataSourceType = strings.Replace(dataSourceType, "/", "-", -1)
-		patterns = append(patterns, patternRoot+dataSourceType+"*,-*-raw,-*-for-merge")
+		patterns = append(patterns, patternRoot+dataSourceType+",-*-raw,-*-for-merge")
```

Comment:
So my concern here is that when you remove that `*` then you may have problems with `github` data source.
See
- if `dataSourceType` is `github/pull_request` then it will return: `sds-fixture-slug-github-pull_request,-*-raw,-*-for-merge` and this is file github/PR will map to PR index
- if `dataSourceType` is `github/issue` then it will return: `sds-fixture-slug-github-issue,-*-raw,-*-for-merge` and this is file github/PR will map to issue index.
- if `dataSourceType` is `github/repository` then it will return: `sds-fixture-slug-github-repository,-*-raw,-*-for-merge` and this is file github/PR will map to repository index.

The same was the case when `*` was added because there was nothing else after that `-github-pull_request` or `-github-issue` or `-github-repository`.

But `dataSourceType` can also be just `github` (without `/category` part)- in this case you will get `sds-fixture-slug-github,-*-raw,-*-for-merge`. This will not hit *ANY* index while it should hit all GitHub indices in this case, this is why `*` did the job. In theory, every data source type can have a sub-type (called category) so in general, you have `dataSourceName` or `dataSourceName/categoryName` - this is only used in GitHub currently.

The problem with the star was when one data source name (like `git`) is a prefix of another data source name (like `github`) in this case `*` makes `git` data source matching all GitHub ones and the git one.

I would implement something similar to your approach but we must handle that `git` should  only match `git` index, `github` should match 3 github indices (gh-issue, gh-pr, gh-repo) and `github/category` should match `github-category`) - just removing `*` will break a plain `github` - eventually, check if data source name is exactly `github` (without a category) and then add `*`.

Reply:
Done